caikit_computer_vision/modules/segmentation/image_processing.py

```diff
@@ -0,0 +1,281 @@
+from caikit.core.exceptions import error_handler
+import alog
+
+log = alog.use_channel("IMAGE_SEGMENTATION")
+error = error_handler.get(log)
+
+
+def maskrcnn_postprocess(
+    results: tuple,
+    uuid: str,
+    image_id: int = 1,
+    threshold=0.2,
+) -> dict:
+    """Postprocess task for MaskRCNN type models.
+
+    Args:
+        results (dict): COCO json from an inference task
+        uuid (str): Unique identifier of image
+        image_id (int, optional): Image ID to store in the task. Defaults to 1.
+
+    Returns:
+        dict: Formatted COCO JSON in OCL format
+    """
+    if len(results) == 0:
+        return {}
+    image_size = results["IMAGES_SIZE"]
+    instances = _convert_results_dict_to_instances(results)
+    log.info(f"Processing Instances of len {len(instances)}")
+    try:
+        result_instances = detector_postprocess(instances, image_size[0], image_size[1])
+    except IndexError as e:
+        log.error(f"Length of MASKS {len(results['MASKS'])}")
+        log.error("No instances found, setting empty annotations", exc_info=e)
+        result_instances = []
+    coco_instances = instances_to_coco_json(result_instances, image_id)
+    coco_json = _assemble_images_coco(
+        file_name=f"{uuid}.jpg",
+        height=image_size[0],
+        width=image_size[1],
+        image_id=image_id,
+    )
+    inf_coco = detectron2_to_coco(
+        detectron2_out=coco_instances,
+        inf_coco=coco_json,
+        score_threshold=threshold,
+        save_coco="",
+    )
+    coco_json = inf_coco.dataset
+    return coco_json
+
+
+def _convert_results_dict_to_instances(results: dict) -> "Instances":
+    """Helper function to convert output of model to Instances.
+
+    Args:
+        results (dict): Raw COCO Dict from Detectron2.
+
+    Returns:
+        Instances: Instances object.
+    """
+    boxes: Boxes = Boxes(results["BOXES"])
+    classes: torch.Tensor = torch.as_tensor(results["CLASSES"])
+    masks: torch.Tensor = torch.as_tensor(results["MASKS"])
+    scores: torch.Tensor = torch.as_tensor(results["SCORES"])
+    image_size = results["IMAGES_SIZE"]
+    instances = Instances(
+        image_size=image_size,
+        pred_boxes=boxes,
+        scores=scores,
+        pred_classes=classes,
+        pred_masks=masks,
+    )
+    return instances
+
+
+def _assemble_images_coco(
+    file_name: str,
+    height: int,
+    width: int,
+    image_id: int = 1,
+) -> "COCO":
+    """Assembles coco dictionary into file and returns COCO object.
+
+    Args:
+        file_name (str): File name
+        height (int): Height of image
+        width (int): Width of image
+        image_id (int, optional): Optional image id. Defaults to 1.
+
+    Returns:
+        COCO: COCO Object
+    """
+    coco_json = dict(
+        images=[
+            {
+                "id": image_id,
+                "file_name": file_name,
+                "height": height,
+                "width": width,
+                "license": 1,
```

Comment:
Just to clarify - the `licenses` aren't defined anywhere are they? Is license ID 1 referring to something specific, or is it just an unsigned int value to make funcs processing this format happy?